deploy/03_Deploy_RandomNumberConsumer.js

```diff
@@ -1,37 +1,41 @@
-const { getNamedAccounts, deployments, network } = require("hardhat")
+const { network } = require("hardhat")
 const {
   networkConfig,
   developmentChains,
   VERIFICATION_BLOCK_CONFIRMATIONS,
 } = require("../helper-hardhat-config")
-const { autoFundCheck, verify } = require("../helper-functions")
+const { verify } = require("../helper-functions")
 
 module.exports = async ({ getNamedAccounts, deployments }) => {
   const { deploy, get, log } = deployments
   const { deployer } = await getNamedAccounts()
   const chainId = network.config.chainId
   let linkTokenAddress
   let vrfCoordinatorAddress
-  let additionalMessage = ""
+  let subscriptionId
 
   if (chainId == 31337) {
     linkToken = await get("LinkToken")
-    VRFCoordinatorMock = await get("VRFCoordinatorMock")
+    VRFCoordinatorV2Mock = await ethers.getContract("VRFCoordinatorV2Mock")
+
+    vrfCoordinatorAddress = VRFCoordinatorV2Mock.address
     linkTokenAddress = linkToken.address
-    vrfCoordinatorAddress = VRFCoordinatorMock.address
-    additionalMessage = " --linkaddress " + linkTokenAddress
+
+    subscriptionId = await VRFCoordinatorV2Mock.callStatic.createSubscription()
+    const fundAmount = networkConfig[chainId]["fundAmount"]
+    await VRFCoordinatorV2Mock.createSubscription()
+    await VRFCoordinatorV2Mock.fundSubscription(subscriptionId, fundAmount)
   } else {
+    subscriptionId = process.env.VRF_SUBSCRIPTION_ID
```

Comment:
I think this makes more sense to add to the `networkConfig` in `helper-hardhat-config` as opposed to an environment variable. We probably could refactor a few environment variables into the networkconfig. The subscriptionId isn't really a sensitive matter.

Reply:
@PatrickAlphaC I was thinking about this a lot, this isn't sensitive indeed. However, I don't know my `subscriptionId` until I create the subscription on [Subscription Management Page](https://vrf.chain.link/). So I come up with two options.

- **Option A**: Create subscription -> find the `helper-hardhat-config` file in the project -> add `subscriptionId` as property to the right js object -> run `yarn hardhat deploy --network <network>`
- **Option B**: Create subscription -> run `VRF_SUBSCRIPTION_ID=<subscriptionId> yarn hardhat deploy --network <network>`

In my opinion, the second way is easier and there is less room for error. Curious to hear your thoughts on this.

Additionally, we can't open some kind of "test subscription" for devs who just quickly want to try VRF and hardcode the `subscriptionId` in `helper-hardhat-config` because after the deployment the consumer contract needs to be manually added as a consumer by the EOA which created the "test subscription" in the first place, and we can't automate this process efficiently imo.

Reply:
Great points. I think you're right!